astropy/io/fits/hdu/image.py

```diff
@@ -246,7 +246,7 @@ def data(self):
 
     @data.setter
     def data(self, data):
-        if 'data' in self.__dict__:
+        if 'data' in self.__dict__ and self.__dict__['data'] is not None:
```

Comment:
Isn't `_data_replaced` set in the `else` block, on line 256 in this case? If it's here the `self.__dict__['data'] is data` test will be skipped as unnecessary. I think it's correct this way but I have no preference.

As a side note, I wonder why `_data_replaced` is set to True just after initialization.
```
im = fits.ImageHDU()
im._data_replaced
Out[14]: True
```